django_mri/models/scan.py

```diff
@@ -147,10 +134,16 @@ class Scan(TimeStampedModel):
         on_delete=models.SET_NULL,
     )
 
+    # #: Associates this scan with some session of a subject.
+    session = models.ForeignKey(
+        "django_mri.Session", blank=True, null=True, on_delete=models.CASCADE,
+    )
+
```

Comment:
It appears this change can be problematic according to the way we implemented this model.

Reply:
Is it possible that moving the call to `self.session.save()` within `save()` to before the `super().save()` call could resolve it?

Reply:
I don't think it would solve the problem, because it means that a scan must be created with a session, and we would like to infer the session during the save method.